pkg/scheduler/factory/factory.go

```diff
@@ -576,7 +579,7 @@ func (c *configFactory) getPluginArgs() (*PluginFactoryArgs, error) {
 		StatefulSetLister:              c.statefulSetLister,
 		NodeLister:                     &nodeLister{c.nodeLister},
 		PDBLister:                      c.pdbLister,
-		NodeInfo:                       &predicates.CachedNodeInfo{NodeLister: c.nodeLister},
+		NodeInfo:                       c.nodeInfo,
```

Comment:
Why not passing `schedulerCache`? It does not look correct to me to add `nodeInfo` to `configFactory`.

Reply:
I update the PR address the comment, but I noticed `configFactory` pass `podLister` instead of `schedulerCache` into `PluginFactoryArgs`, do I need to fix this too?

```go
func (c *configFactory) getPluginArgs() (*PluginFactoryArgs, error) {
	return &PluginFactoryArgs{
		PodLister:                      c.podLister, // could use c.schedulerCache
		ServiceLister:                  c.serviceLister,
		ControllerLister:               c.controllerLister,
		ReplicaSetLister:               c.replicaSetLister,
		StatefulSetLister:              c.statefulSetLister,
		NodeLister:                     &nodeLister{c.nodeLister},
		PDBLister:                      c.pdbLister,
		NodeInfo:                       c.schedulerCache,
		PVInfo:                         &predicates.CachedPersistentVolumeInfo{PersistentVolumeLister: c.pVLister},
		PVCInfo:                        &predicates.CachedPersistentVolumeClaimInfo{PersistentVolumeClaimLister: c.pVCLister},
		StorageClassInfo:               &predicates.CachedStorageClassInfo{StorageClassLister: c.storageClassLister},
		VolumeBinder:                   c.volumeBinder,
		HardPodAffinitySymmetricWeight: c.hardPodAffinitySymmetricWeight,
	}, nil
}
```

Reply:
We can consider that change too, but don't make that change in this PR.

Reply:
I've opened an issue to track this potential change https://github.com/kubernetes/kubernetes/issues/79538

Reply:
shouldn't we also change NodeLister to use the scheduler's node cache as well?

Reply:
> shouldn't we also change NodeLister to use the scheduler's node cache as well?

We could do that in other PR if necessary.

Reply:
can you please explain why is it necessary to change NodeInfo to use the scheduler cache, but NodeLister may not be?

Reply:
> In the scheduler, we use the cache to provide a globally consistent view of the cluster, even if that view is slightly out of date. Data read from the core NodeInfo may not have been processed by the scheduler cache yet. Letting data "skip ahead" like this can cause stale info to get persisted in the scheduler cache, and other subtle bugs. https://github.com/kubernetes/kubernetes/issues/75617#issue-424410497


This PR is supposed to make the predicates read node info from scheduler cache, and updates `NodeLister` is irrelevant to this purpose, so I'd prefer to do it in another one.

Reply:
I am not asking to do it in this PR. What I am asking is whether or not the problem is limited to what the issue claims to be. Anyways, I will take a closer look and open another issue, thanks for doing this one.

Reply:
@ahg-g Thank you for your clarifying, we could track the related issue in #79538

Reply:
ok, I got a chance to look more into this, I think changing NodeLister to use the scheduler cache is more than a possible clean up issue, which #79538 declares to be.

NodeLister is used in [CalculateInterPodAffinityPriority](https://github.com/kubernetes/kubernetes/blob/38e8b8a9f65e66b779a0985e17868b036c1d601b/pkg/scheduler/algorithmprovider/defaults/register_priorities.go#L73) along with NodeInfo. NodeInfo now is not reading from the same source as NodeLister, and so potentially this change could have solved one thing, but caused another.

I opened #79629 to track this issue.